### ISSUE TITLE
URL-encoded parts of a copied text are URL-decoded

### ISSUE THREAD

**Issue by swergas** [zeroclipboard/zeroclipboard#565]
I want to use zeroclipboard to copy an URL which contains URL-encoded characters like %2F and %3A. But once I click copy, the version of the URL which gets saved in my computer's clipboard is a URL-decoded version of it. For example if I use http://zeroclipboard.org/#demo to copy-paste "%2F", I get "/" instead.

**Comment by JamesMGreene** [comment]
Are you using ZeroClipboard `v2.2.0`?

Have you verified this outside of the ZeroClipboard demo? If not, it could just be a display issue there.

**Comment by swergas** [comment]
Yes, I'm using ZeroClipboard v2.2.0 in a project, that's where I experienced the problem first.

**Comment by JamesMGreene** [comment]
Ugh, that could be a big problem for lots of folks. :disappointed:

**Comment by JamesMGreene** [comment]
P.S. Most likely a bug in Flash's encode/decode utilities for URIs. Flash, as a whole, absolutely stinks at encoding/decoding things correctly (including JSON... and apparently URIs, too!).

**Comment by JamesMGreene** [comment]
As expected, the JS parts appear to work:

![image](https://cloud.githubusercontent.com/assets/417751/7236193/f9c852ba-e758-11e4-9560-796cd7ff7c08.png)

So yes... likely another Flash encoding/decoding deficiency to look into further.

**Comment by JamesMGreene** [comment]
@swergas Can you still reproduce this issue?

I am no longer able to reproduce this on my Mac OS X system for any browser (Chrome, Firefox, Opera, Safari).  I am assuming this must have been fixed in a more recent version of Flash Player (my versions were `21.x` and `23.x`).

It's a shame we didn't make note of what version of Flash either of us were using at the time.  I recall reproducing this in my Chrome on Mac OS X at the time, which would've been Chrome 41. Looks like Flash Player was _probably_ version `17.x` at the time but could've possibly been `16.x`.

**Comment by swergas** [comment]
No I can't reproduce it anymore :) It works fine when I try to copy "%2F" on http://zeroclipboard.org/#demo

**Comment by swergas** [comment]
Hmmm sorry, I tested also on Ubuntu+Firefox and I still have the bug.

**Comment by JamesMGreene** [comment]
@swergas: I'm assuming your Ubuntu has the latest NPAPI Flash version for Linux (which is like 5 years old), version 11?

If that's the case, I may just end up punting on this issue in favor of the HTML Clipboard API (see #171 and #650) being the best choice for Linux browsers going forward.

Additionally, you can now apparently proxy the latest versions of the PPAPI Flash into an NPAPI adapter on Linux, which I suspect would also resolve the issue for Linux users: http://www.pcworld.com/article/2993902/browsers/how-to-get-the-latest-version-of-flash-on-firefox-for-linux-after-adobes-abandonment.html

**Comment by JamesMGreene** [comment]
@swergas: I'm assuming your Ubuntu has the latest NPAPI Flash for Linux (which is like 5 years old), version `11.2.x`?

If that's the case, I may just end up punting on this issue in favor of the HTML Clipboard API (see #171 and #650) being the best choice for Linux browsers going forward.

Additionally, you can now apparently also either:
 1. [Update your NPAPI Flash for Linux to version `23.x` by utilizing the Beta channel](http://blogs.adobe.com/flashplayer/2016/08/beta-news-flash-player-npapi-for-linux.html#sthash.iEBLTX9C.dpbs). And yes, it jumps from `11.2.x` to `23.x`. 😆 
 2. Proxy the latest versions of the PPAPI Flash into an NPAPI adapter on Linux using [Fresh Flash Player](http://www.pcworld.com/article/2993902/browsers/how-to-get-the-latest-version-of-flash-on-firefox-for-linux-after-adobes-abandonment.html)

**Comment by JamesMGreene** [comment]
Yeah, I verified it is definitely still a problem with Flash 11.2 on Linux Firefox (using SauceLabs).

`ZeroClipboard.state()` looks like:

```json
{
    "browser": {
        "userAgent": "Mozilla/5.0 (X11; Linux i686; rv:45.0) Gecko/20100101 Firefox/45.0",
        "platform": "Linux i686",
        "appName": "Netscape"
    },
    "flash": {
        "version": "11.2.202",
        "pluginType": "netscape",
        "disabled": false,
        "outdated": false,
        "sandboxed": false,
        "unavailable": false,
        "degraded": false,
        "deactivated": false,
        "overdue": false,
        "ready": true
    },
    "zeroclipboard": {
        "version": "2.2.0",
        "config": {
            "swfPath": "http://zeroclipboard.org/javascripts/zc/v2.2.0/ZeroClipboard.swf",
            "trustedDomains": [
                "zeroclipboard.org"
            ],
            "cacheBust": true,
            "forceEnhancedClipboard": false,
            "flashLoadTimeout": 30000,
            "autoActivate": true,
            "bubbleEvents": true,
            "containerId": "global-zeroclipboard-html-bridge",
            "containerClass": "global-zeroclipboard-container",
            "swfObjectId": "global-zeroclipboard-flash-bridge",
            "hoverClass": "zeroclipboard-is-hover",
            "activeClass": "zeroclipboard-is-active",
            "forceHandCursor": false,
            "title": null,
            "zIndex": 999999999
        }
    }
}
```

**Comment by JamesMGreene** [comment]
I'm going to defer this until after I've integrated the HTML Clipboard API support.  If that resolves the issue for current versions of the browsers, I'll just document this problem (and try to identify which versions of Flash it affects) rather than attempting to workaround Flash's serialization bugs.
